overlord/configstate/configcore/coredump.go

```diff
@@ -0,0 +1,121 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/gadget/quantity"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/sysconfig"
+)
+
+const (
+	optionCoredumpEnable     = "system.coredump.enable"
+	optionCoredumpMaxuse     = "system.coredump.maxuse"
+	coreOptionCoredumpEnable = "core." + optionCoredumpEnable
+	coreOptionCoredumpMaxuse = "core." + optionCoredumpMaxuse
+
+	coredumpCfgSubdir = "coredump.conf.d"
+	coredumpCfgFile   = "ubuntu-core.conf"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations[coreOptionCoredumpEnable] = true
+	supportedConfigurations[coreOptionCoredumpMaxuse] = true
+}
+
+func validMaxUseSize(sizeStr string) error {
+	if sizeStr == "" {
+		return nil
+	}
+
+	_, err := quantity.ParseSize(sizeStr)
```

Comment:
Should we retain the actual size and re-render to the format that systemd-coredumpd expects? Right now we kind of depend (very subtly so) on the fact that we attempt to handle the same quantities the same way. Should systemd change or our side change to accept something that is not also accepted by systemd, I think it would be better to parse and re-render the value.

Reply:
Currently what we accept in `ParseSize` is a subset of what is accepted by systemd MaxUse parameters, so the output will be always something acceptable by systemd. We do similar things for tmp and in other places. If we change things on our side we should asses usages of ParseSize everywhere, otherwise each option should do its own parsing.